test/e2e/lib/pages/frontend/comment-likes-component.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { By } from 'selenium-webdriver';
+
+/**
+ * Internal dependencies
+ */
+import AsyncBaseContainer from '../../async-base-container';
+import * as driverHelper from '../../driver-helper';
+
+export default class CommentLikesComponent extends AsyncBaseContainer {
+	/**
+	 * Encapsulates post comment like clicks and assertions
+	 *
+	 * @param driver WebDriver driver
+	 * @param comment Text used to locate the comment under test
+	 * @param url Url of the post where the comment can be found
+	 */
+	constructor( driver, comment, url ) {
+		super( driver, By.xpath( `//div[@class='comment-content']/p[.='${ comment }']` ), url );
+		this.comment = comment;
+	}
+
+	static async Expect( driver, comment ) {
+		const page = new this( driver, comment );
+		await page._expectInit();
+		return page;
+	}
+
+	static async Visit( driver, comment, url ) {
+		const page = new this( driver, comment, url );
+		await page._visitInit();
+		return page;
+	}
+
+	async likeComment() {
+		const commentLikeLink = By.xpath(
+			`//div[@class='comment-content']/p[.='${ this.comment }']/../p/a[@class='comment-like-link']`
+		);
+		await driverHelper.scrollIntoView( this.driver, commentLikeLink, 'end' );
+		await driverHelper.clickWhenClickable( this.driver, commentLikeLink );
+	}
+	async expectLiked() {
+		const commentLikedText = By.xpath(
+			`//div[@class='comment-content']/p[.='${ this.comment }']/../p/span[starts-with(text(),'Liked by')]`
+		);
+		await driverHelper.waitTillPresentAndDisplayed( this.driver, commentLikedText );
```

Comment:
Needs a rebase as `waitTillPresentAndDisplayed` was [recently refactored](https://github.com/Automattic/wp-calypso/pull/51506) to `waitUntilLocatedAndVisible`.